include/flatbuffers/base.h

```diff
@@ -173,5 +173,163 @@ inline size_t PaddingBytes(size_t buf_size, size_t scalar_size) {
   return ((~buf_size) + 1) & (scalar_size - 1);
 }
 
+// Helper method that retrieves ::data() from a vector in a way that is
+// compatible with pre C++11 STLs (e.g stlport).
+template <typename T>
+inline T *VectorData(std::vector<T> *vector) {
+#if defined(FLATBUFFERS_CPP98_STL)
+  return &((*vector)[0]);
+#else
+  return vector->data();
+#endif  // defined(FLATBUFFERS_CPP98_STL)
+}
+
+template <typename T>
+inline const T *VectorData(const std::vector<T> *vector) {
+#if defined(FLATBUFFERS_CPP98_STL)
+  return &((*vector)[0]);
+#else
+  return vector->data();
+#endif  // defined(FLATBUFFERS_CPP98_STL)
 }
+
+template <typename T>
+inline void VectorEmplaceBack(std::vector<T> *vector, const T &data) {
+#if defined(FLATBUFFERS_CPP98_STL)
+  vector->push_back(data);
+#else
+  vector->emplace_back(data);
+#endif  // defined(FLATBUFFERS_CPP98_STL)
+}
+
+#if !(defined(_MSC_VER) && _MSC_VER <= 1700 /* MSVC2012 */)
+#ifndef FLATBUFFERS_CPP98_STL
+template <typename T>
+using is_scalar = std::is_scalar<T>;
+template <typename T>
+using is_floating_point = std::is_floating_point<T>;
+template <typename T>
+using is_unsigned = std::is_unsigned<T>;
+#else
+// Map C++ TR1 templates defined by stlport.
+template <typename T>
+using is_scalar = std::tr1::is_scalar<T>;
+template <typename T>
+using is_floating_point = std::tr1::is_floating_point<T>;
+template <typename T>
+using is_unsigned = std::tr1::is_unsigned<T>;
+#endif  // FLATBUFFERS_CPP98_STL
+#else
+// MSVC 2010 doesn't support C++11 aliases.
+template <typename T>
+struct is_scalar : public std::is_scalar<T> {};
+template <typename T>
+struct is_floating_point : public std::is_floating_point<T> {};
+template <typename T>
+struct is_unsigned : public std::is_unsigned<T> {};
+#endif  // !(defined(_MSC_VER) && _MSC_VER <= 1700 /* MSVC2012 */)
+
+#ifndef FLATBUFFERS_CPP98_STL
+#if !(defined(_MSC_VER) && _MSC_VER <= 1700 /* MSVC2012 */)
+template <class T>
+using unique_ptr = std::unique_ptr<T>;
+#else
+// MSVC 2010 doesn't support C++11 aliases.
```

Comment:
This forwards unique_ptr into the flatbuffers namespace, I've updated the comment to describe the rationale.